Dependencies/UrlLib/Source/Unix/UrlRequest.cpp

```diff
@@ -54,18 +72,58 @@ namespace UrlLib
 
         gsl::span<const std::byte> ResponseBuffer() const
         {
-            throw std::runtime_error{"Not impemented"};
+            return m_responseBuffer;
         }
 
     private:
 
+        static void Append(std::string& string, char* buffer, size_t nitems)
+        {
+            string.insert(string.end(), buffer, buffer + nitems);
+        }
+
+        static void Append(ByteArray& byteArray, char* buffer, size_t nitems)
+        {
+            auto bytes = reinterpret_cast<std::byte const*>(buffer);
+            byteArray.insert(byteArray.end(), bytes, bytes + nitems);   
+        }
+
+        template<typename DataT> void LoadFile(DataT& data)
+        {
+            auto curl = curl_easy_init();
+            if (curl)
+            {
+                data.clear();
+                curl_easy_setopt(curl, CURLOPT_URL, m_url.data());
+                curl_easy_setopt(curl, CURLOPT_FOLLOWLOCATION, 1L);
+
+                curl_write_callback callback = [](char* buffer, size_t size, size_t nitems, void* userData) {
+                    auto& data = *static_cast<DataT*>(userData);
+                    Append(data, buffer, nitems);
+                    return nitems;
+                };
+
+                curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, callback);
+                curl_easy_setopt(curl, CURLOPT_WRITEDATA, &data);
+
+                auto result = curl_easy_perform(curl);
+                if (result != CURLE_OK)
+                {
+                    throw std::exception();
+                }
+
+                curl_easy_cleanup(curl);
+            }
+        }
```

Comment:
this is synchronous. we should try to implement this correctly now.

Reply:
https://github.com/BabylonJS/BabylonNative/issues/232